packages/jellyfish-api-core/__tests__/category/loan/tokenSplits.test.ts

```diff
@@ -0,0 +1,966 @@
+  await testing.generate(1)
+}
+
+describe('Token splits', () => {
+  beforeEach(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterEach(async () => {
+    await testing.container.stop()
+  })
+
+  it('should check token after split', async () => {
+    const tokenTSLA = await testing.rpc.token.getToken(tslaID)
+    const newMintedTSLA = tokenTSLA[tslaID].minted.multipliedBy(2)
+
+    // set a token split at current block + 2 for TSLA and FB
+    const splitBlock = await testing.rpc.blockchain.getBlockCount() + 2
+    await testing.rpc.masternode.setGov({
+      ATTRIBUTES: {
+        [`v0/oracles/splits/${splitBlock}`]: `${tslaID}/2`
+      }
+    })
+    await testing.generate(2)
+    await checkTokenSplit(tslaID, 'TSLA', '/v1', newMintedTSLA, true, false)
+  })
+
+  it('should check active vault after split', async () => {
+    const tokenFB = await testing.rpc.token.getToken(fbID)
+    const newMintedFB = tokenFB[fbID].minted.multipliedBy(2)
+
+    const vaultFBBefore = await testing.rpc.vault.getVault(vaultId4) as VaultActive
+
+    // set a token split at current block + 2 for FB
+    const splitBlock = await testing.rpc.blockchain.getBlockCount() + 2
+    await testing.rpc.masternode.setGov({
+      ATTRIBUTES: {
+        [`v0/oracles/splits/${splitBlock}`]: `${fbID}/2`
+      }
+    })
+    await testing.generate(2)
+    const newFBID = await checkTokenSplit(fbID, 'FB', '/v1', newMintedFB, true, false)
+
+    {
+      // check vaults
+      const vaultFBAfter = await testing.rpc.vault.getVault(vaultId4) as VaultActive
+      expect(vaultFBAfter.vaultId).toStrictEqual(vaultFBBefore.vaultId)
+      expect(vaultFBAfter.ownerAddress).toStrictEqual(vaultFBBefore.ownerAddress)
+      expect(vaultFBAfter.state).toStrictEqual(VaultState.FROZEN)
+      expect(vaultFBAfter.collateralAmounts).toStrictEqual(vaultFBBefore.collateralAmounts)
+      const loanAmountBefore = Number(vaultFBBefore.loanAmounts[0].replace('@FB', ''))
+      const loanAmountAfter = Number(vaultFBAfter.loanAmounts[0].replace('@FB', ''))
+      expect(loanAmountAfter).toBeGreaterThan(loanAmountBefore * 2) // interest accumilation before split
+      // check newFBID is still locked.
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${newFBID}`]).toStrictEqual('true')
+
+      // try to update the vault
+      const promise = testing.rpc.vault.depositToVault({ vaultId: vaultId4, from: collateralAddress, amount: '10@DFI' })
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow('RpcApiError: \'Test DepositToVaultTx execution failed:\nFixed interval price currently disabled due to locked token\', code: -32600, method: deposittovault')
+    }
+
+    {
+      // unlock the token and check
```

Comment:
hmm.. so `unlock` must be done manually??
the token will be locked every after split??

Reply:
can unlock manually once all looks good. not sure if defid has an automatic job to do this after some time. haven't seen any.
yes

Reply:
yea i'm good on this ady
>  The lock will last 3 hours before the split and 3 hours after the split. After the lock is lifted, vaults will be halted for 1 hour before they become operational again.
https://blog.defichain.com/amzn-stock-split/